microcosm_pubsub/result.py

```diff
@@ -154,8 +154,18 @@ def log(self, logger, opaque):
         )
 
     def error_reporting(self, sentry_enabled, opaque):
-        if not sentry_enabled:
+        if not all([
+            sentry_enabled,
+            self.result in [
+                MessageHandlingResultType.FAILED,
+                MessageHandlingResultType.EXPIRED,
```

Comment:
Casing this late in the flow seems weird. What's the reason?

Reply:
Not sure what you mean by casing. In terms of position in flow, these result types should behave as normal for everything else in this class. It's only the sentry integration that doesn't care about the other result types.